frontend/js/paradigm.js

```diff
@@ -0,0 +1,159 @@
+
+  const nextParadigmSize = getNextParadigmSize(paradigmSize)
+
+  toggleButton.addEventListener('click', () => {
+    displayButtonAsLoading(toggleButton)
+
+    fetch(window.Urls['cree-dictionary-paradigm-detail']() + `?lemma-id=${lemmaId}&paradigm-size=${nextParadigmSize}`).then(r => {
+      if (r.ok) {
+        return r.text()
+      } else {
+        // TODO: show error on the loading component
+        throw new Error(`${r.status} ${r.statusText} when loading paradigm: ${r.text()}`)
+      }
+    }).then(
+      text => {
+        const newParadigm = document.createRange().createContextualFragment(text)
+
+        // TODO: the backend should SOLELY maintain this:
+        if (mostDetailedParadigmSizeIsSelected()) {
+          setParadigmSizeToggleButtonText('-', 'show less')
+        } else {
+          setParadigmSizeToggleButtonText('+', 'show more')
+        }
+
+        window.history.replaceState({}, document.title, updateQueryParam('paradigm-size', nextParadigmSize))
+
+        const paradigmContainer = document.getElementById('paradigm')
+        paradigmContainer.querySelector('.js-replaceable-paradigm').replaceWith(newParadigm)
+
+        paradigmSize = nextParadigmSize
+        setupParadigmSizeToggleButton(paradigmSize)
+
+        function setParadigmSizeToggleButtonText(symbol, text) {
+          newParadigm.querySelector('.js-button-text').textContent = text
+          newParadigm.querySelector('.js-plus-minus').textContent = symbol
+        }
+      }
+    ).catch((err) => {
+      displayButtonAsError(toggleButton)
+      console.error(err)
+    })
+  })
+
+  // TODO: the backend should know this:
+  function mostDetailedParadigmSizeIsSelected() {
+    return ALL_PARADIGM_SIZES.indexOf(nextParadigmSize) === ALL_PARADIGM_SIZES.length - 1
+  }
+}
+
+/**
+ * Make the button look like it's loading.
+ */
+function displayButtonAsLoading(toggleButton) {
+  toggleButton.classList.add('paradigm__size-toggle-button--loading')
+}
+
+/**
+ * Make the button look like something went wrong.
+ */
+function displayButtonAsError(toggleButton) {
+  toggleButton.classList.remove('paradigm__size-toggle-button--loading')
+  // TODO: should have an error state for the toggle button!
+}
+
+
+/**
+ * cycles between BASIC, FULL, LINGUISTIC
+ *
+ * @param {String} size
+ * @return {String}
+ */
+function getNextParadigmSize(size) {
+  return ALL_PARADIGM_SIZES[(ALL_PARADIGM_SIZES.indexOf(size) + 1) % ALL_PARADIGM_SIZES.length]
+}
+
+///////////////////////////// Internal functions /////////////////////////////
+
+/**
+ * Read JSOn data produced by Django's `json_script` filter during HTML template generation
+ */
+function readDjangoJsonScript(id) {
+  const jsonScriptElement = document.getElementById(id)
+  if (jsonScriptElement) {
+    return JSON.parse(jsonScriptElement.textContent)
+  } else {
+    return undefined
+  }
+}
+
+/**
+ * https://stackoverflow.com/a/11654596
```

Comment:
But but but 😢 … can’t we use `URLSearchParams` like https://stackoverflow.com/a/11654596 instead of this line noise ???

Reply:
Oops meant https://stackoverflow.com/a/41542008/14558 was distracted by trying and failing to come up with a sufficient expression of horror at that … perl … monstrosity

Reply:
Thank you!